src/mono/wasm/runtime/loader/config.ts

```diff
@@ -3,24 +3,25 @@
 
 import BuildConfiguration from "consts:configuration";
 import type { DotnetModuleInternal, MonoConfigInternal } from "../types/internal";
-import type { DotnetModuleConfig } from "../types";
+import type { DotnetModuleConfig, MonoConfig } from "../types";
 import { ENVIRONMENT_IS_WEB, exportedRuntimeAPI, loaderHelpers, runtimeHelpers } from "./globals";
-import { initializeBootConfig, loadBootConfig } from "./blazor/_Integration";
-import { BootConfigResult } from "./blazor/BootConfig";
-import { BootJsonData } from "../types/blazor";
 import { mono_log_error, mono_log_debug } from "./logging";
 import { invokeLibraryInitializers } from "./libraryInitializers";
 import { mono_exit } from "./exit";
 
 export function deep_merge_config(target: MonoConfigInternal, source: MonoConfigInternal): MonoConfigInternal {
+    // If source has collection fields set to null (produced by boot config for example), we should maintain the target values
     const providedConfig: MonoConfigInternal = { ...source };
-    if (providedConfig.assets) {
+    if (providedConfig.assets !== undefined) {
         providedConfig.assets = [...(target.assets || []), ...(providedConfig.assets || [])];
     }
-    if (providedConfig.environmentVariables) {
+    if (providedConfig.resources !== undefined) {
```

Comment:
`normalizeConfig` should set `resources` to `{}` or even setup empty structure, so that other code would not have to check it later.

also we need deepmerge for `resources` ?

Reply:
Probably yes